### PR TITLE
release raiwidgets and responsibleai 0.16.0

### PR DESCRIPTION
**Pull request by imatiach-msft**
## v0.16.0

- educational materials
  - major readme refactoring changes + filename changes for public release of RAI Dashboard
  - add missing doc for pred_y_dataset and min_child_samples
  - add npm and yarn installation instructions
- new features
  - ## Counterfactual
    - counterfactual list style updates: highlight counterfactual examples columns with different color, and bold size
    - highlight first and last row of counterfactual list
    - implement list method for counterfactual manager
  - ## Error Analysis
    - add scrollbar area around matrix cells when there are many categories
    - add warning message for downsampled explanation data in ErrorAnalysis
- bug fixes and tests
  - ## Responsible AI Dashboard
    - add e2e UI tests for Dataset Explorer - whisker plot
    - add e2e UI tests for Dataset Explorer - scatter plot
    - add e2e UI tests for Dataset Explorer - flyout
    - add e2e UI tests for Dataset Explorer - cohort scenarios
    - add e2e UI tests for Model Statistics - chart tests
    - refactor binding this with arrow func
    - add error message for duplicate cohort name
    - handle context cohort delete cases for data explorer
  - ## RAIInsights
    - throw error on mutating predict function in RAIInsights
    - use spilt orientation to save train and test dataframe to address duplicate indices condition in them
    - add .json extension to dtypes files and add tests
    - add predict and predict_proba outputs to RAIInsights save method
    - refactor RAIInsights load function into smaller functions
    - refactor classifier detection logic into common function
    - fix issue with missing directories
    - gracefully handle failure in loading model.pkl from disk
    - remove unused test dependencies responsibleai
    - add missing unit tests for \_convert_to_list()
    - separate out save() and load() tests
    - add more unit tests for serialize_json_safe()
    - update responsibleai to interpret-community 0.23.0
  - ## Counterfactual
    - add e2e UI tests for what-if counterfactuals - creating counterfactuals
  - ## Error Analysis
    - fix coverage calculation for cell tooltips in heatmap for recall, precision and f1 score metrics
    - fix handling numpy types in heatmap for categories
  - ## Interpret
    - improve state management for explainer manager
    - test for save/load/save bug in Explanations
- other
  - fix top broken image link in tour notebook on github
  - add isort to build pipelines
  - fix SubBarChart flaky test
  - update notebook tests for new notebooks
  - add flake8-bugbear for better python code contructs
  - add flake8-breakpoint to avoid code checkin with active breakpoints
  - remove csv files from source for cognitive services example notebooks
  - add yarn e2e-watch shortcut
  - replace existing user token with shared account token for releases

### CODE REVIEW

File: responsibleai/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 dice-ml>=0.7.2,<0.8
 econml~=0.12.0
 jsonschema
-erroranalysis>=0.1.28
+erroranalysis>=0.1.29
```
Comment (riedgar-ms) [inline responsibleai/requirements.txt:4]: Should this be a separate PR?

Comment (imatiach-msft) [reply responsibleai/requirements.txt:4]: it's just updating to latest error analysis package

Comment (riedgar-ms) [reply responsibleai/requirements.txt:4]: Discussed over PM.

**Review by riedgar-ms — APPROVED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -24,6 +24,67 @@ this file to understand what changed.
...
+    - refactor binding this with arrow func
+    - add error message for duplicate cohort name
+    - handle context cohort delete cases for data explorer
+  - ## RAIInsights
+    - throw error on mutating predict function in RAIInsights
+    - use spilt orientation to save train and test dataframe to address duplicate indices condition in them
+    - add .json extension to dtypes files and add tests
+    - add predict and predict_proba outputs to RAIInsights save method
+    - refactor RAIInsights load function into smaller functions
+    - use getattr to reference predict_proba
```
Comment (gaugup) [inline CHANGES.md:61]: I think this PR was edited to refactor certain logic for detecting if model is classifier. https://github.com/microsoft/responsible-ai-toolbox/pull/1075 (Refactor classifier detection logic into common function)

**Review by gaugup — CHANGES_REQUESTED** [review]

File: CHANGES.md (hunk 2 above)
Comment (imatiach-msft) [reply CHANGES.md:61]: updated!

File: CHANGES.md [hunk 3]
```diff
@@ -24,6 +24,67 @@ this file to understand what changed.
...
 
+## v0.16.0
+
+- educational materials
+  - major readme refactoring changes + filename changes for public release of RAI Dashboard
+  - add missing doc for pred_y_dataset and min_child_samples
+  - add npm and yarn installation instructions
+  - add romanlutz to CODEOWNERS of raiwidgets
+  - update NOTICE file
+  - add pull request template
```
[the remark this answers is not in the document]
Comment (imatiach-msft) [reply CHANGES.md:35]: ok, removed these:
  - add romanlutz to CODEOWNERS of raiwidgets
  - update NOTICE file
  - add pull request template

File: CHANGES.md [hunk 4]
```diff
@@ -24,6 +24,67 @@ this file to understand what changed.
...
+- new features
+  - ## Counterfactual
+    - counterfactual list style updates: highlight counterfactual examples columns with different color, and bold size
+    - highlight first and last row of counterfactual list
+    - implement list method for counterfactual manager
+  - ## Error Analysis
+    - add scrollbar area around matrix cells when there are many categories
+    - add warning message for downsampled explanation data in ErrorAnalysis
+- breaking changes
```
[the remark this answers is not in the document]
Comment (imatiach-msft) [reply CHANGES.md:44]: done!
